lib/logstash/inputs/gelf.rb

```diff
@@ -90,12 +113,84 @@ def stop
   rescue IOError # the plugin is currently shutting down, so its safe to ignore theses errors
```

Comment:
Dosen't this need to be if tcp and if udp?

Otherwise when stopping it will try to stop udp and not tcp if tcp is enabled.

Reply:
I'm extremely confused by this stop method. It is used in line 195:

`while !stop?
line, client = @udp.recvfrom(8192)`

But stop does @udp.close 
How can you receive messages from a closed socket? I'm new to ruby. What am I missing?

Reply:
/me is new too :). So not sure. I just presumed we needed to do an if check for both udp and tcp